src/index.ts

```diff
@@ -100,30 +101,32 @@ function VitePluginRestart(options: Options = {}): Plugin {
         ...restartGlobs,
         ...reloadGlobs,
       ])
-      server.watcher.on(
-        'change',
-        (file) => {
-          if (micromatch.isMatch(file, restartGlobs)) {
-            timerState = 'restart'
-            schedule(() => {
-              touch(configFile)
-              console.log(
-                c.dim(new Date().toLocaleTimeString())
-                + c.bold(c.blue(' [plugin-restart] '))
-                + c.yellow(`restarting server by ${pathPlatform.relative(root, file)}`),
-              )
-              timerState = ''
-            })
-          }
-          else if (micromatch.isMatch(file, reloadGlobs) && timerState !== 'restart') {
-            timerState = 'reload'
-            schedule(() => {
-              server.ws.send({ type: 'full-reload' })
-              timerState = ''
-            })
-          }
-        },
-      )
+      fileEvents.forEach((ev) => {
```

Comment:
Better to move the body to a function and call with `change`, `add`, `unlink` directly

Reply:
Got it, should be ok now